File: Trabajo3/CodigoIndiceDeterminado.java
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        List<Integer> numeros = new ArrayList<>();
        numeros.add(10);
        numeros.add(20);
        numeros.add(30);
        numeros.add(40);
        numeros.add(50);

        int valorAInsertar = 75;
        int indice = 2; // Insertar en la posición 2 (entre 20 y 30)

        if (indice >= 0 && indice <= numeros.size()) {
            numeros.add(indice, valorAInsertar);
        } else {
            System.out.println("Índice fuera de rango.");
        }

        // Mostrar el resultado
        System.out.print("Lista actualizada: ");
        for (int num : numeros) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
} 

File: Trabajo3/CodigoIndiceDeterminado.cpp
#include <iostream>
#include <vector>

int main() {
    std::vector<int> numeros = {10, 20, 30, 40, 50};
    int valorAInsertar = 25;
    int indice = 2; // Insertar en la posición 2 (entre 20 y 30)

    if (indice >= 0 && indice <= numeros.size()) {
        numeros.insert(numeros.begin() + indice, valorAInsertar);
    } else {
        std::cout << "Índice fuera de rango." << std::endl;
    }

    // Mostrar el resultado
    std::cout << "Vector actualizado: ";
    for (int num : numeros) {
        std::cout << num << " ";
    }
    std::cout << std::endl;

    return 0;
}